tests/into_iterator.rs

```diff
@@ -45,3 +45,25 @@ impl ::core::iter::IntoIterator for Numbers3 {
         <Vec<i32> as ::core::iter::IntoIterator>::into_iter(self.numbers)
     }
 }
+
+#[derive(IntoIterator)]
+struct Generic1<T> {
+    #[into_iterator(owned, ref, ref_mut)]
+    items: Vec<T>,
+}
```

Comment:
@MegaBluejay it would be nice to add actual tests for values, to ensure that the generated code produces correct `Iterator`s.

For already existing tests too.

Reply:
Especially for the new tests I think this would be great. If you don't mind doing it for the old tests too that is very much appreciated.